### PR TITLE
Adding to developer guidelines ipfs investigation

### LINKED ISSUE #443: Investigate IPFS
Investigate IPFS tecnology and write a document in the developer guidelines

### PR DESCRIPTION
**Pull request by warnerHurtado**
### What does this PR do?

- resolves #443 

### Steps to test
1. Go to developer guidelines
1. Click on IPFS (Interplanetary File System) in the sidebar
1. See the new doc.

#### CheckList
- [ ] Follow proper Markdown format
- [ ] The content is adequate
- [ ] The content is available in both english and spanish
- [ ] I Ran a spell check

### CODE REVIEW

File: docs/interplanetary-file-system.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+This IPFS protocol has also been implemented in various networks related to the blockchain. This is due to its storage capacity and large number of benefits that it can bring to these networks. Some of the networks that use this storage protocol are: [Filecon](https://filecoin.io/), it is a decentralized storage network, the first blockchain network to implement the IPFS protocol in its network, [Storj](https://www.storj.io/), is another decentralized storage blockchain network that also uses the protocol. Another great use case is with NFTs ([Non Fungible Tokens](https://guias.eoscostarica.io/docs/aprender-eosio/nfts-en-eosio)), this technology is currently trending and is used for encrypted digital assets which uses IPFS for the storage of its data.
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:61]: link to use case is with NFTs seems to be spanish content

File: docs/interplanetary-file-system.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+id: interplanetary-file-system
+title: IPFS (Interplanetary File System)
+sidebar_label: IPFS (Interplanetary File System)
+---
+
+The IPFS (Interplanetary File System) is a decentralized network protocol that is based on a distributed network, this to create a direct relationship with the p2p method ([peer to peer](https://es.wikipedia.org/wiki/Peer-to-peer)), the purpose of this protocol is to store files or information quickly and safely. In this way, the IPFS changes the traditional form of information storage from centralized to decentralized, bringing with it many benefits such as access to information without depending on a single storage point for its respective access, increased storage, control of use of your own data and among many other benefits.
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:7]: Remove "The" at the start of the sentence

Comment (xavier506) [reply docs/interplanetary-file-system.md:7]: refer to IPFS as IPFS not "the IPFS"

In this way, the IPFS...

Comment (xavier506) [reply docs/interplanetary-file-system.md:7]: include a link to the main IPFS page  

https://ipfs.io/

File: docs/interplanetary-file-system.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+The current protocol performs the following steps to perform a search:
+
+- First take the web address and then query the DNS for the IP of the server location.
+- Already with the IP in hand, the server information query is made and the respective download is started.
+- After downloading, it shows us the information of said server.
+
+This is how the traditional protocol works, **content searches by location**, in which it is necessary to know the address to which you want to query the data. In the case of IPFS it is different, as mentioned above, the protocol does this search by content. This system uses the following steps to perform a search:
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:33]: use bold on **search by content** (same as search by location)

File: docs/interplanetary-file-system.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+- It is not necessary to know the address, only what is needed is searched in the system.
+- The system sends the search that is being carried out to the network of nodes. This information is protected by encryption, hash and digital signature, this in order to protect the information and give access only to those who are authorized.
+- After sending the search, the nodes respond with information on the available versions of what was requested.
+- By choosing one of the results obtained, you can see both its content and its information history.
+
+That is the way the IPFS system works, for this reason it is known as **search by content** since we simply indicate what we need and the network of nodes responds with the option to consult the data that is going to be required.
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:40]: i think they use the term "content addressing"

File: docs/interplanetary-file-system.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+## Advantages and disadvantages of IPFS
+
+Some of the advantages are as follows:
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:44]: Some of the advantages of using IPS are:

File: docs/interplanetary-file-system.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+- Decentralized storage so it does not depend on a single place for access and is accessible at any time.
+- It is scalable, so future improvements and functionalities can be added.
+- Fast data storage and query.
+- Greater storage capacity.
+- There is no censorship of content, being decentralized it is impossible for any content to be censored.
+
+Some of its disadvantages:
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:52]: Some of the disadvantages of using IPFS are:

File: docs/interplanetary-file-system.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- It is a new technology that when implemented will bring with it some complications when wanting to run it completely on the internet.
+- It is still in the process of improvement and development.
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:54]: I believe it is currently still in beta

File: docs/interplanetary-file-system.md (hunk 1 above)
Comment (xavier506) [inline docs/interplanetary-file-system.md:61]: Remove "This"

Comment (xavier506) [reply docs/interplanetary-file-system.md:61]: The NFT link is referencing the Spanish site , please point to the english equivalent

File: docs/interplanetary-file-system.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
+---
+id: interplanetary-file-system
+title: IPFS (Interplanetary File System)
+sidebar_label: IPFS (Interplanetary File System)
```
Comment (xavier506) [inline docs/interplanetary-file-system.md:4]: have you checked if this sidebar label fits in the UI?

**Review by xavier506 — APPROVED** [review]
LGTM

File: docs/interplanetary-file-system.md (hunk 2 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md (hunk 3 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md (hunk 4 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md (hunk 5 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md (hunk 6 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+- Decentralized storage so it does not depend on a single place for access and is accessible at any time.
+- It is scalable, so future improvements and functionalities can be added.
+- Fast data storage and query.
+- Greater storage capacity.
+- There is no censorship of content, being decentralized it is impossible for any content to be censored.
+
+Some of its disadvantages:
+- It is a new technology that when implemented will bring with it some complications when wanting to run it completely on the internet.
+- It is still in the process of improvement and development.
+- Being new is difficult to use for some people.
```
[the remark this answers is not in the document]
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md (hunk 1 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md]: Attended.

File: docs/interplanetary-file-system.md (hunk 8 above)
Comment (warnerHurtado) [reply docs/interplanetary-file-system.md:4]: Attended.

**Pull request merged**